### PR TITLE
feat(dynamic-sampling): Add prioritise low volume transaction rules [TET-736]

### PR DESCRIPTION
**Pull request by RaduW**
This PR adds rule generation for low volume transactions

### CODE REVIEW

File: src/sentry/dynamic_sampling/rules/biases/boost_rare_transactions_rule.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                    "inner": [
+                        {
+                            "op": "eq",
+                            "name": "event.transaction",
+                            "value": [name],
+                            "options": {"ignoreCase": True},
+                        }
+                    ],
+                },
+                "id": RESERVED_IDS[RuleType.BOOST_LOW_VOLUME_TRANSACTIONS],
```
Comment (andriisoldatenko) [inline src/sentry/dynamic_sampling/rules/biases/boost_rare_transactions_rule.py:42]: you should increase ID similar to release bias.

File: src/sentry/conf/server.py [hunk 2]
```diff
@@ -890,6 +890,11 @@ def SOCIAL_AUTH_DEFAULT_USERNAME():
         # Run job every 1 hour
         "schedule": crontab(minute=0),
     },
+    "dynamic-sampling-prioritize-transactions": {
+        "task": "sentry.dynamic_sampling.tasks.prioritise_transactions",
+        # Run job every 1 hour
```
Comment (andriisoldatenko) [inline src/sentry/conf/server.py:895]: may be let's run it every 4h?

File: src/sentry/dynamic_sampling/rules/biases/boost_rare_transactions_rule.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            }
+        )
+
+    # add a rule for all other transactions:
```
Comment (andriisoldatenko) [inline src/sentry/dynamic_sampling/rules/biases/boost_rare_transactions_rule.py:46]: i don't get this one,

File: src/sentry/dynamic_sampling/rules/biases/boost_rare_transactions_rule.py (hunk 1 above)
Comment (RaduW) [reply src/sentry/dynamic_sampling/rules/biases/boost_rare_transactions_rule.py:42]: yes, good catch

File: src/sentry/dynamic_sampling/rules/biases/boost_rare_transactions_rule.py (hunk 3 above)
Comment (RaduW) [reply src/sentry/dynamic_sampling/rules/biases/boost_rare_transactions_rule.py:46]: This is a left over from the initial algorithm which could increase the sampling rate for unspecified transactions.
This is not the case anymore... will remove this.
Thanks.

**Review by iambriccardo — CHANGES_REQUESTED** [review]

**Review by andriisoldatenko — APPROVED** [review]

**Review by iambriccardo — APPROVED** [review]

File: src/sentry/dynamic_sampling/rules/biases/boost_rare_transactions_rule.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        )
+        ret_val: List[Rule] = []
+
+        if len(transaction_map) == 0:
+            return ret_val  # no point returning any rules the project rule should take over
+
+        idx = 0
+        for name, transaction_rate in transaction_map.items():
+            if sample_rate != 0:
+                transaction_rate /= sample_rate
```
Comment (iambriccardo) [inline src/sentry/dynamic_sampling/rules/biases/boost_rare_transactions_rule.py:38]: nit: I would add a comment explaining the rationale behind this formula.

**Pull request merged**